docs/guide/build.md

```diff
@@ -43,6 +43,20 @@ module.exports = defineConfig({
 
 For example, you can specify multiple Rollup outputs with plugins that are only applied during build.
 
+## Chunking Strategy
+
+You can configure how chunks are split using `build.rollupOptions.manualChunks` (see [Rollup docs](https://rollupjs.org/guide/en/#outputmanualchunks)). Until Vite 2.7, the default chunking strategy divided the chunks into `index` and `vendor`. It is a good strategy for some SPAs, but it is hard to provide a general solution for every Vite target use case. From Vite 2.8, `manualChunks` is no longer modified by default. You can continue to use the Split Vendor Chunk strategy by adding the `splitVendorChunkPlugin` in your config file:
+
+```js
+// vite.config.js
+import { splitVendorChunkPlugin } from 'vite'
+module.exports = defineConfig({
+  plugins: [splitVendorChunkPlugin()]
+})
+```
+
+This strategy is also provided as a `splitVendorChunk({ cache: SplitVendorChunkCache )` factory, in case composition with custom logic is needed. `cache.reset()` needs to be called at `buildStart` for build watch mode to work correctly in this case.
```

Comment:
Idk what the use case for custom `reset` is, but we could add an `onReset` callback instead of using a custom class.

Reply:
The cache used by `splitVendorChunk` needs to be cleaned up on every `buildStart`. I don't understand the `onReset` callback, how would that work?

Reply:
So the plugin would call `cache.clear` on the Map object and then call `onReset` callback after that in case there's any custom logic needed. But as I said, I have zero idea why anyone would need custom reset logic here. Perhaps you have an idea? 😄

Reply:
Oh here's a use case. A plugin that wraps the `splitVendorChunk` plugin may need to be reset its own cache.

Reply:
If the user is already doing a new plugin, I think it is better that they use the function `splitVendorChunk` and they can reset their caches at the same time as resetting the `SplitVendorChunkCache` (on `buildStart`). I don't think we need to have an API for the plugin. The plugin is there so users can add in a single line the old strategy, and we are exposing the low level function because it was requested several times before to compose it with other logic.

Reply:
Okay that's fair, but I'm still confused why the `SplitVendorChunkCache` class is necessary if you can just call `cache.clear` on the Map object.

Reply:
Oh right, you said:
> I don't know if we will need to expand the cache later. I wanted to expose the cache for the method, not a particular Map that is an implementation detail and may change

That doesn't really make sense to me. If we decide not to use a Map in the future (not sure why we would tbh), then we can handle that then.

But if you really believe the cache class is the right move, I'll defer to your judgment, since this isn't a huge deal or anything. :)

Reply:
It just feels to me that we are leaking an implementation detail if we share the Map. In this case it isn't even something useful like a map of modules, it is an internal cache used by the algorithm. I also don't like having to create a class, but it feels safer here to avoid the need for a breaking change.